src/plugins/usage_collection/server/collector/stats_collector.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Collector } from './collector';
+
+/**
+ * Used to declare a collector used for stats gathering instead of usage.
+ * @private
+ */
+export class StatsCollector<TFetchReturn, ExtraOptions extends object = {}> extends Collector<
+  TFetchReturn,
+  ExtraOptions
+> {
```

Comment:
Just FMI, what is the purpose of sub classing `Collector` here? Do we plan on increasing / diverging their API at some point?

Reply:
It comes from this request by @TinaHeiligers: https://github.com/elastic/kibana/issues/83522. I guess it's just to be more explicit. If you think it's not needed, I'm happy to close the PR.

For now, we only have Stats and Usage Collectors. And the only difference is that Usage collectors require the `schema` definition, while Stats don't. And sometimes, we only need to collect the information reported by Usage Collectors (leaving Stats aside). There used to be more differences in the past. But after decoupling from Monitoring, they were removed.

Reply:
> If you think it's not needed, I'm happy to close the PR

I lack the functional knowledge to have an opinion right now tbh. If @TinaHeiligers think it's better, I wouldn't disagree 😄

Reply:
@afharo This is one of those situations where we're adapting code for a _possible_ future implementation. I still think it's good to have but it might be a little premature since we don't have a need for it right now.
I'll leave it up to your judgment to decide on merging or not since you now have a better idea of where this might go in the future.

Reply:
I don't have a strong opinion on this either. I understand the benefit of being more explicit with the types. Collectors are registered via the factory methods `makeStatsCollector` and `makeUsageCollector`. So it _makes sense_ each method calls a class with that name in it.

That said, it's more of a readability thing over an actual benefit (for now). Maybe we can benefit from it if we ever filter by `instanceof StatsCollector` or their behaviours vary. It feels like it creates more confusion (readability improvement failed), and we can do `!instanceof UsageCollector` to get the `StatsCollector`s if required. I'll go ahead and close the PR and issue for now. We can revisit it in the future if needed.